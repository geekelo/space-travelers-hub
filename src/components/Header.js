import { NavLink } from 'react-router-dom';
import styles from '../styles/styles.module.css';
import '../styles/activestyles.css';
import Logo from '../assets/planet.png';

const Header = () => (
  <div className={`${styles.container_fluid} ${styles.header}`}>
    <nav className={`${styles.container} ${styles.navbar}`}>
      <p className={styles.brand}>
        <img src={Logo} alt="logo" />
        <span>Space Travel&apos;s Hub</span>
      </p>
      <div className={`${styles.menusection}`}>
        <p><NavLink to="/" className={`${styles.menuitems} menuitems`}>Rocket</NavLink></p>
        <p><NavLink to="/missions" className={`${styles.menuitems} menuitems`}>Missions</NavLink></p>
      </div>
    </nav>
  </div>
);

export default Header;
