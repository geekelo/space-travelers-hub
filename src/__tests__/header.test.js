import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';

describe('Header Component', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <Router>
        <Header />
      </Router>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
