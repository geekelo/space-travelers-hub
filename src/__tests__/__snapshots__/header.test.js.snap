// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container_fluid header"
  >
    <nav
      class="container navbar"
    >
      <p
        class="brand"
      >
        <img
          alt="logo"
          src="planet.png"
        />
        <span>
          Space Travel's Hub
        </span>
      </p>
      <div
        class="menusection"
      >
        <p>
          <a
            aria-current="page"
            class="menuitems menuitems active"
            href="/"
          >
            Rocket
          </a>
        </p>
        <p>
          <a
            class="menuitems menuitems"
            href="/missions"
          >
            Missions
          </a>
        </p>
        <p>
          <a
            class="menuitems myprofileText menuitems"
            href="/myprofile"
          >
            My Profile
          </a>
        </p>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
